js/blocks/components/fetch-input.js

```diff
@@ -13,44 +13,29 @@ import classNames from 'classnames';
  * WordPress dependencies
  */
 import { __, sprintf, _n } from '@wordpress/i18n';
-import { Component, createRef } from '@wordpress/element';
+import { createRef, useEffect, useState } from '@wordpress/element';
 import { decodeEntities } from '@wordpress/html-entities';
 import { UP, DOWN, ENTER } from '@wordpress/keycodes';
 import { BaseControl, Spinner, withSpokenMessages, Popover } from '@wordpress/components';
 import { withInstanceId } from '@wordpress/compose';
 import { addQueryArgs } from '@wordpress/url';
-const { apiFetch } = wp;
+import apiFetch from '@wordpress/api-fetch';
 
 // Since FetchInput is rendered in the context of other inputs, but should be
 // considered a separate modal node, prevent keyboard events from propagating
 // as being considered from the input.
 const stopEventPropagation = ( event ) => event.stopPropagation();
 
-class FetchInput extends Component {
-	/**
-	 * Constructs the component class.
-	 */
-	constructor( { autocompleteRef } ) {
-		super( ...arguments );
-
-		this.onBlur = this.onBlur.bind( this );
-		this.onFocus = this.onFocus.bind( this );
-		this.onChange = this.onChange.bind( this );
-		this.onKeyDown = this.onKeyDown.bind( this );
-		this.autocompleteRef = autocompleteRef || createRef();
-		this.inputRef = createRef();
-		this.updateSuggestions = this.updateSuggestions.bind( this );
-		this.setInputValidity = this.setInputValidity.bind( this );
-
-		this.suggestionNodes = [];
-
-		this.state = {
-			loading: false,
-			results: [],
-			showSuggestions: false,
-			selectedSuggestion: null,
-		};
-	}
+const FetchInput = ( props ) => {
```

Comment:
Here's the class refactored to be a function with hooks.

Reply:
There's a unit test for the component:

```npm install && npm run test:js fetch-input`

Reply:
The [test-js Travis job](https://travis-ci.com/getblocklab/block-lab/jobs/285249839) passed 5 times in a row, which makes me a little less concerned that these tests will be flaky. That was something I noticed with E2E tests before.